lib/dateTime.js

```diff
@@ -1167,3 +1167,38 @@ export function formatTimeFromTimestamp(timestamp, options) {
 	const date = parseLocalDateTimeFromTimestamp(timestamp);
 	return formatTime(date, options);
 }
+
+export function formatRelativeDateTime(date, { numeric = 'always', style = 'long' } = {}) {
+
+	const now = new Date();
+	const then = new Date(date);
+
+	if (!Intl.RelativeTimeFormat) {
+		return now.toDateString() === then.toDateString() ? formatTime(then) : formatDate(then);
```

Comment:
I'm not sure what's best, since the context could affect things. For instance, the UI might say "Updated 20 minutes ago" or "Updated 2 days ago" and it sounds like the fallback would be "Updated 4:31 PM" or "Updated Jan 8, 2023" , respectively. The "today" case leaves us with awkward phrasing since it's gone from being a duration to being an actual date or time.

On the other hand, [browser support is pretty good](https://caniuse.com/mdn-javascript_builtins_intl_relativetimeformat) so very few users should see this fallback, right? If so, I'm good with this approach. It keeps the length minimal too, which might be important depending on the layout where this is used.

Reply:
Yeah browser support is really good... we don't have recent data anymore but I'd guess probably 95% of our users.

Your example of "Updated {relativeTime}" is a good one to remember, as we'll want to try and support something like that in the auto-updating, `<abbr>`-based web component wrapper around this. Either that, or maybe it's better to treat these relative time terms as special "rich" formatting (like bold or a link) within lang terms -- which is another thing we want to support soon. Need to think more about this.